### PR TITLE
Building an Always-On CPU Profiler

### PR DESCRIPTION
**Pull request by htroisi**

### CODE REVIEW

File: content/author.yaml [hunk 1]
```diff
@@ -32,3 +32,12 @@
   bio: Lead SWE @ New Relic, Founding engineer @ Pixie Labs
   twitter: 'https://twitter.com/philkuznets'
   email: 'philkuz@pixielabs.ai'
+
+- id: Omid Oazizi
```
Comment (oazizi000) [inline content/author.yaml:36]: Azizi

File: content/blog/cpu-profiling/index.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+---
+path: '/cpu-profiling'
+title: 'Building an Always-On CPU Profiler'
+date: 2021-5-24T06:00:00.000+00:00
+featured_image: hero-image.png
+categories: ['Pixie Team Blogs']
+authors: ['Omid Oazizi', 'Pete Stevenson']
+emails: ['omid@pixielabs.ai', 'pete@pixielabs.ai']
```
Comment (oazizi000) [inline content/blog/cpu-profiling/index.md:8]: oazizi@pixielabs.ai

File: content/blog/cpu-profiling/index.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+---
+path: '/cpu-profiling'
+title: 'Building an Always-On CPU Profiler'
+date: 2021-5-24T06:00:00.000+00:00
+featured_image: hero-image.png
+categories: ['Pixie Team Blogs']
+authors: ['Omid Oazizi', 'Pete Stevenson']
```
Comment (oazizi000) [inline content/blog/cpu-profiling/index.md:7]: Omid Azizi

File: content/author.yaml [hunk 4]
```diff
@@ -32,3 +32,12 @@
...
   email: 'philkuz@pixielabs.ai'
+
+- id: Omid Azizi
+  bio: Principal Engineer @ New Relic, Founding engineer @ Pixie Labs
+  twitter: 'https://twitter.com/oazizi'
+  email: 'oazizi@pixielabs.ai'
+
+- id: Pete Stevenson
+  bio: Principal Engineer @ New Relic, Founding engineer @ Pixie Labs
+  email: 'pete@pixielabs.ai'
```
Comment (oazizi000) [inline content/author.yaml:43]: jps@pixielabs.ai

File: content/blog/cpu-profiling/index.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+A third approach to profiling is simulation, as used by Valgrind/Callgrind. Valgrind uses no instrumentation, but runs your program through a virtual machine which profiles the execution. This approach provides a lot of information, but is also high in overheads.
+
+The table below summarizes properties of a number of popular performance profilers.
+
+| Profiler | Methodology | Deployment | Traces Libraries/System Calls? | Performance Overhead
+| --- | --- | --- | --- | --- |
+| [gprof](https://sourceware.org/binutils/docs/gprof/) | Instrumentation + Sampling | Recompile & Rerun | No | High (up to [260%](https://www.researchgate.net/publication/221235356_Low-overhead_call_path_profiling_of_unmodified_optimized_code)) |
+| [Callgrind](https://valgrind.org/docs/manual/cl-manual.html) | Simulation | Rerun | Yes? | Very High ([>400%](https://www.cs.cmu.edu/afs/cs.cmu.edu/project/cmt-40/Nice/RuleRefinement/bin/valgrind-3.2.0/docs/html/cl-manual.html)) |
```
Comment (oazizi000) [inline content/blog/cpu-profiling/index.md:80]: Remove the question mark on Yes?

**Review by oazizi000 — COMMENTED** [review]
Still don't know how to preview the final version. Want to take a look at that before the final approve.

File: content/blog/cpu-profiling/index.md (hunk 5 above)
Comment (htroisi) [reply content/blog/cpu-profiling/index.md:80]: I removed the question mark for both Callgrind and gperftools.

**Review by oazizi000 — APPROVED** [review]

**Review by oazizi000 — APPROVED** [review]

**Review by oazizi000 — APPROVED** [review]

**Review by zasgar — APPROVED** [review]

**Pull request merged**